### ISSUE TITLE
Documentation Incomplete: audio_output_fifo_pull_sample(...)

### ISSUE THREAD

**Issue by ahogen** [xmos/lib_tsn#27]
1. This documentation block (shown below) is inconstant with the actual function.
2. Why are all functions/tasks defined in `.c` or `.xc` *except* for this one which is in `.h`? Please be consistent.
3. There is a `static inline unsigned int` declaration of this function in the corresponding .c file, with no implementation defined. I've removed it in my own code without any issues. Why is it there?

```
/**
 *  \brief Used by the audio output system to pull the next sample from the FIFO
 *
 *  If there are no samples in the buffer, a zero will be returned. The current
 *  ref clock time is passed into the function, and the FIFO will record this
 *  time if the sample which has been removed was the marked sample
 *
 *  \param s the FIFO to remove a sample from
 *  \param timestamp the ref clock time of the sample playout
 */
 /*
unsigned int
audio_output_fifo_pull_sample(buffer_handle_t s,
                                  unsigned int timestamp);
*/
__attribute__((always_inline))
unsafe static inline unsigned int
audio_output_fifo_pull_sample(buffer_handle_t s0,
                              unsigned index,
                              unsigned int timestamp)
```

Applies to:
- [/src/audio_buffering/audio_output_fifo.h](https://github.com/xmos/lib_tsn/blob/master/lib_tsn/src/audio_buffering/audio_output_fifo.h)
- [/src/audio_buffering/audio_output_fifo.c](https://github.com/xmos/lib_tsn/blob/master/lib_tsn/src/audio_buffering/audio_output_fifo.c)

**Comment by larry-xmos** [comment]
Thanks for reporting the outdated comment, I will repair it

audio_output_fifo_pull_sample is in a header file for performance reasons. With the way XC compiler currently handles inlining, we found we had to use the always_inline attribute to get it to always inline into audio_buffer_manager as needed.

The empty inline definition must be there as a convention. You would typically write that to give yourself a non-inlined version of the function to link against later should the compiler decide that the inline version isn't used at compile time and discard it. The empty definition can be removed, as you found.
